src/formatDate.js

```diff
@@ -8,7 +8,32 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(fromFormat[3]);
+  const dateObj = {};
+
+  for (let i = 0; i < 3; i++) {
+    dateObj[fromFormat[i]] = dateParts[i];
+  }
+
+  if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
+    dateObj['YY'] = dateObj['YYYY'].slice(-2);
```

Comment:
You can directly use 'YY' instead of creating a new key in the `dateObj`. Since the 'YYYY' key will not be used after this conversion, it's safe to overwrite it: `dateObj['YYYY'] = dateObj['YYYY'].slice(-2);`